interfaces/builtin/kvm.go

```diff
@@ -38,8 +48,60 @@ const kvmConnectedPlugAppArmor = `
 
 var kvmConnectedPlugUDev = []string{`KERNEL=="kvm"`}
 
+type kvmInterface struct {
+	commonInterface
+}
+
+var procCpuinfo = "/proc/cpuinfo"
+
+func getCpuFlags() (flags string, err error) {
+	buf, err := ioutil.ReadFile(procCpuinfo)
+	if err != nil {
+		// if we can't read cpuinfo, we want to know _why_
+		return "", fmt.Errorf("error: %v", err)
+	}
+
+	pattern := []byte("\nflags\t\t:")
+	idx := bytes.LastIndex(buf, pattern)
+	if idx >= 0 {
+		offset := idx + len(pattern)
+		endidx := bytes.Index(buf[offset:], []byte("\n"))
+		return string(buf[offset : endidx+offset]), nil
+	}
+
+	// if not found (which will happen on non-x86 architectures, which is ok
+	// because they'd typically not have the same info over and over again),
+	// return whole buffer; otherwise, return from just after the \n
+	return string(buf[idx+1:]), nil
+}
+
+func (iface *kvmInterface) KModConnectedPlug(spec *kmod.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
+	// Check CPU capabilities to load suitable module
+	// TODO: this only considering x86 CPUs, but some ARM, PPC and S390 CPUs also support KVM
```

Comment:
Please at the very least support x86_64, i386 and amd64. I don't know of any easily available ARMv7 boards that have kvm (none of the reference platforms do) so I think it is fine to skip that.

Reply:
I need to double-check i386 (think it uses same cpu flag and kernel module), but x86_64 and amd64 are covered.

Reply:
vmx is available on some i386. Happy to skip the others. Comment updated to be clearer